mkosi/util.py

```diff
@@ -269,3 +269,7 @@ def mandatory_variable(name: str) -> str:
         return os.environ[name]
     except KeyError:
         die(f"${name} must be set in the environment")
+
+
+def dir_only_contains(dir: Path, tgt: PathString) -> bool:
+    return not dir.is_dir() or (dir / tgt).exists() and len(list(dir.glob("*"))) == 1
```

Comment:
Thinking about this some more, would it make sense for this the directory to also be skipped if it's empty?

Reply:
> Thinking about this some more, would it make sense for this the directory to also be skipped if it's empty?

It would, but... is realistic to think of a system with a `/usr/lib/modules/<kver>` directory with nothing within it? As a curiosity, some yeas ago I had to add a check with `/usr/lib/modules/<kver>/kernel` in dracut to fix a "real" customer case...

https://github.com/openSUSE/dracut/commit/24aa3f7488b0d668902c0d2bc2566e4fdc289ff4

Reply:
> It would, but... is realistic to think of a system with a /usr/lib/modules/<kver> directory with nothing within it? 

That's why I wondered. I mean the Debian system I just checked, (that's long-running and has been updating for over a year now, so quite different from the images we build, has about 20 directories, more than three quarters o which are empty.